lib/charms/opensearch/v0/constants_secrets.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""In this file we declare the constants and enums used by TLS related components."""
+
+# The unique Charmhub library identifier, never change it
+LIBID = "2f539a53ab0a4916957beaf1d6b27124"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+# Secret labels
+SECRET_LABEL_PREFIX = "charmsecret"
+SECRET_LABEL_SEPARATOR = "-000-"
```

Comment:
I am _really_ unhappy with this.

However, https://github.com/canonical/operator/blob/90cb5dfc5ea5fdfa0aacfc405fa30b805a7c6951/ops/model.py#L1082 

```
Keys should be lowercase letters and digits, at least 3 characters long, start with a letter, and not start or end with a hyphen.
```
...doesn't leave us much room for creativity (like using `tls/certs/ca-cert` or alike :-( )

Reply:
Since we moved from secret keys (that have strict restriction on characters) to secret labels, we can use `:` in the name.
Thus it's now `[app|unit]:<app/unit-name>:<secret-name>`

Reply:
On second thoughts, to be on the safe side (and adding support for multiple related applications, the format should be as follows: 
```
opensearch:app:admin-pwd
opensearch:unit:0:tls

where:
    - opensearch: app_name
    - app / unit: scope of secret
    - 0: unit_id in case of unit secret
    - tls / admin-pwd: secret label
```

Reply:
Done.